solutions/cses-1202.mdx

```diff
@@ -5,86 +5,99 @@ title: Investigation
 author: Andrew Wang
 ---
 
-**Time Complexity:** $\mathcal{O}(E\log V)$
+## Explanation
+
+We can run Dijkstra keeping track of
+- the distance: $\texttt{dist}[]$
+- the number of ways with the minimum distance: $\texttt{num}[]$
+- the minimum flights with the minimum distance: $\texttt{minf}[]$, and 
+- the maximum flights with the minimum distance: $\texttt{maxf}[]$.
 
-We can run Dijkstra keeping track of the distance: $dist[]$, the number of ways
-with the minimum distance: $num[]$, the minimum flights with the minimum
-distance: $minf[]$, and the maximum flights with the minimum distance: $maxf[]$.
 For every node, $v$, we take into consideration all of its neighbors, $u$. If we
 can reach $u$ in a shorter distance than its current minimum, we update the
-distance and reset $num[u]$, $minf[u]$, and $maxf[u]$. We also have to take into
+distance and reset $\texttt{num}[u]$, $\texttt{minf}[u]$, and $\texttt{maxf}[u]$.
+
+We also have to take into
 consideration if we can reach $u$ in an equivalent distance. If so, we update:
 
 $$
-num[u] = num[v] + num[u]
+\texttt{num}[u] = \texttt{num}[v] + \texttt{num}[u]
 $$
 
 $$
-minf[u] = min(minf[u], minf[v] + 1)
+\texttt{minf}[u] = \min(\texttt{minf}[u], \texttt{minf}[v] + 1)
 $$
 
 $$
-maxf[u] = max(maxf[u], maxf[v] + 1)
+\texttt{maxf}[u] = \max(\texttt{maxf}[u], \texttt{maxf}[v] + 1)
 $$
 
+## Implementation
+
+**Time Complexity:** $\mathcal{O}((N + M)\log N)$
+
+<LanguageSection>
+	
+<CPPSection>
+
 ```cpp
 #include <bits/stdc++.h>
 using namespace std;
 
 typedef long long ll;
-vector<pair<ll, int>> edge[100001];
-ll dist[100001]; ll num[100001];
-int minf[100001];int maxf[100001];
-bool v[100001];
-ll MAX = 1000000000000000L; ll MOD = 1000000007L;
-void djikstra(int s){
+
+const int MAXN = 100001;
+const ll MAX = 0x3f3f3f3f3f3f3f3f;
```

Comment:
what the hell is this

Reply:
[take a look at this](https://stackoverflow.com/a/18429250).

Reply:
> is a repetition of equal bytes, which means you can easily memset stuff to INF;

Reply:
w8 yo, why can't we just make `MAX` as `LLONG_MAX`?

Reply:
maybe overflow? this is still a really weird number though

Reply:
how could it be overflow if we're storing it in a `long long`?

Reply:
idk, ask the author or something

Reply:
it doesn't overflow

Reply:
> how could it be overflow if we're storing it in a `long long`?

lol `long long`s could still overflow

Reply:
> > how could it be overflow if we're storing it in a `long long`?
> 
> lol `long long`s could still overflow

oh i was referring to how `LLONG_MAX` won't overflow if it's stored in a `long long` :laughing: .

Reply:
> maybe overflow? this is still a really weird number though

he's talking about `0x3f3f3f3f3f3f3f3f`, I think.